crates/core/src/sql/compiler.rs

```diff
@@ -14,6 +14,7 @@ use spacetimedb_vm::expr::{ColumnOp, CrudExpr, DbType, Expr, QueryExpr, SourceEx
 use spacetimedb_vm::operator::OpCmp;
 
 /// Compile the `SQL` expression into a `ast`
+#[tracing::instrument(skip(db, tx))]
 pub fn compile_sql(db: &RelationalDB, tx: &MutTxId, sql_text: &str) -> Result<Vec<CrudExpr>, DBError> {
```

Comment:
This function has 2 major steps: `compile_to_ast` for parsing the `sql` and `compile_statement` to turn that into our `ast`.  I think is redundant to have  `compile_to_ast` here or if wanna see the cost then add the loop that call `compile_statement`.